### DISCUSSION TITLE
Why does my api route become a prerender route?

### DISCUSSION THREAD

**Discussion by sknightq** [vercel/next.js#61699] [App Router]
I'm using the App Folder with NextJS 13. 
I created 2 route.ts in arena folder like following:
![image](https://github.com/vercel/next.js/assets/8033144/7ac669c4-bb5d-4eb4-99b2-44dcd613ae3f)
The **/api/arena/rank** path (a GET http request) became a prerender route, which makes the result of api is always same.
What's more, the api will output the result of query DB (I print console.log) when I run ```npm run build```.

However, the output is missing and the api request become dynamically if I move the GET handle into the /api/arena folder.
![image](https://github.com/vercel/next.js/assets/8033144/2a59e382-2b1c-419e-8ab0-7f0b36f84122)

Do I miss anything rules about handle route? The /api/arena route like follwing:
![image](https://github.com/vercel/next.js/assets/8033144/5ed98467-5125-47eb-b7e7-53d810e9af34)

**Comment by icyJoseph** [comment]
Hi,

Route handler get statically optmized, when certain conditions are met:

- https://nextjs.org/docs/app/building-your-application/routing/route-handlers#behavior

The above link explains it in detail, but shortly:

> Route Handlers are cached by default when using the GET method with the Response object.

And you can OPT out from that:

- Using the Request object with the GET method.
- Using any of the other HTTP methods.
- Using [Dynamic Functions](https://nextjs.org/docs/app/building-your-application/routing/route-handlers#dynamic-functions) like cookies and headers.
- The [Segment Config Options](https://nextjs.org/docs/app/building-your-application/routing/route-handlers#segment-config-options) manually specifies dynamic mode.
